frontend/device/android/device.lua

```diff
@@ -510,6 +511,7 @@ function Device:_showLightDialog()
     elseif action == C.ALIGHTS_DIALOG_CANCEL then
         logger.dbg("Dialog Cancel, brightness: " .. self.powerd.fl_intensity)
         self.powerd:setIntensityHW(self.powerd.fl_intensity)
+        self.powerd:_decideFrontlightState()
```

Comment:
I would possibly move that one inside Android's `setIntensityHW` implementation. (We do the same on a few other platforms, e.g., Kobo).